packages/flutter/lib/src/widgets/selection_area.dart

```diff
@@ -0,0 +1,196 @@
+  void initState() {
+    super.initState();
+    _gestureRecognizers[PanGestureRecognizer] = GestureRecognizerFactoryWithHandlers<PanGestureRecognizer>(
+      () => PanGestureRecognizer(supportedDevices: <PointerDeviceKind>{PointerDeviceKind.mouse}),
+      (PanGestureRecognizer instance) {
+        instance
+          ..onDown = _handleDragDown
+          ..onStart = _handleDragStart
+          ..onUpdate = _handleDragUpdate
+          ..onCancel = _cancelSelection
+          ..dragStartBehavior = DragStartBehavior.down;
+      },
+    );
+    _gestureRecognizers[TapGestureRecognizer] = GestureRecognizerFactoryWithHandlers<TapGestureRecognizer>(
+        () => TapGestureRecognizer(debugOwner: this, supportedDevices: <PointerDeviceKind>{PointerDeviceKind.mouse}),
+        (TapGestureRecognizer instance) {
+          instance.onTap = _cancelSelection;
+        },
+      );
+  }
+
+  @override
+  void dispose() {
+    for (final Selectable selectable in _selectables) {
+      selectable.clear();
+    }
+    _selectables.clear();
+    super.dispose();
+  }
+
+  @override
+  void add(Selectable selectable) {
+    _selectables.add(selectable);
+  }
+
+  @override
+  void remove(Selectable selectable) {
+    _selectables.remove(selectable);
+    selectable.clear();
+  }
+
+
+  void _handleDragDown(DragDownDetails details) {
+    _cancelSelection();
+  }
+
+  void _handleDragStart(DragStartDetails details) {
+    final Offset offset = (context.findRenderObject() as RenderBox?)!.localToGlobal(details.localPosition);
+    _startSelection(offset);
+  }
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    final Offset offset = (context.findRenderObject() as RenderBox?)!.localToGlobal(details.localPosition);
+    _updateSelection(offset);
+  }
+
+  void _startSelection(Offset offset) {
+    _start = offset;
+  }
+
+  void _updateSelection(Offset offset) {
+    _end = offset;
+    final Rect globalSelectionRect = Rect.fromPoints(_start!, _end!);
+    for (final Selectable selectable in _selectables) {
+      selectable.update(globalSelectionRect);
+    }
+  }
+
+  void _cancelSelection() {
+    _start = null;
+    _end = null;
+    for (final Selectable selectable in _selectables)
+      selectable.clear();
+  }
+
+  void _selectAll(Intent intent) {
+    _cancelSelection();
+    _startSelection(Offset.zero);
+    _updateSelection(Offset.infinite);
+  }
+
+  Future<void> _copy(Intent intent) async {
+    final List<Object> selections = <Object>[];
+    for (final Selectable selectable in _selectables) {
+      final Object? data = selectable.copy();
+      if (data != null)
+        selections.add(data);
+    }
+    if (selections.isEmpty)
+      return;
+    // The order in which these should be concatenated is ???. For now this
+    // sorts by the order in which the selectables were added.
+    final String data = selections.join('\n');
+    Clipboard.setData(ClipboardData(text: data));
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
@gspencergoog I feel like we talked about this before, but I'm unsure why I can't get these shortcuts to fire. i suppose I need the change that allows shortcut keys to fallback to parent shortcuts?